### PR TITLE
Wait for AddressIndexer initialization

### PR DESCRIPTION
**Pull request by quantumagi**
See https://stratisplatformuk.visualstudio.com/STRAX/_workitems/edit/5233/.

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs [hunk 1]
```diff
@@ -600,6 +605,12 @@ public AddressBalancesResult GetAddressBalances(string[] addresses, int minConfi
         /// <inheritdoc />
         public VerboseAddressBalancesResult GetAddressIndexerState(string[] addresses)
         {
+            // If the containing feature is not initialized then wait a bit.
+            if (!this.storeSettings.AddressIndex)
+                throw new NotSupportedException("Address indexing is not supported.");
```
Comment (zeptin) [inline src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs:610]: Maybe 'not enabled' would be more informative than not supported

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs]: Ditto what @zeptin  said 👍

File: src/Stratis.Bitcoin/Builder/Feature/FullNodeFeature.cs [hunk 2]
```diff
@@ -61,5 +75,22 @@ public virtual void Dispose()
...
+
+        public void WaitInitialized()
+        {
+            if (!this.IsEnabled())
+                throw new NotSupportedException($"{this.GetType()} is not enabled.");
+
+            // If this feature is awaiting initialization then wait a bit.
+            while (this.State == FeatureInitializationState.Uninitialized || this.State == FeatureInitializationState.Initializing)
+            {
+                Thread.Sleep(1000);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/Builder/Feature/FullNodeFeature.cs:92]: Can we put the 1000 in a constant? I would even wait 3 to 5 seconds?

**Review by fassadlr — COMMENTED** [review]
Just those 2 small changes I reckon. Great PR, We've been to move away from the `string` State for some time!

**Review by fassadlr — APPROVED** [review]

**Pull request merged**